tests/unit/igraph_steiner_tree_fpt.c

```diff
@@ -0,0 +1,119 @@
+
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.h"
+
+int main()
+{
+   // printf("Starting Tests\n");
+    igraph_t g_empty, g_lm;
+
+    igraph_vector_int_t steiner_terminals,steiner_terminals_null;
+    igraph_vector_t weights_empty, weights_lm;
+    
+    igraph_vector_init(&weights_empty, 0);
+
+    igraph_vector_int_init(&steiner_terminals, 4);
+    igraph_vector_int_init(&steiner_terminals_null, 0);
+
+    VECTOR(steiner_terminals)[0] = 0;
+    VECTOR(steiner_terminals)[1] = 1;
+    VECTOR(steiner_terminals)[2] = 2;
+    VECTOR(steiner_terminals)[3] = 3;
```

Comment:
Tip: initializing a vector to a given value is more easily done with `igraph_vector_int_init_int()` or `igraph_vector_int_init_real()`. For example, `igraph_vector_int_init_int(&steiner_terminals, 4 /* element count */, 0, 1, 2, 3);` You can grep the contets of the `tests` directory for example uses of these functions.